include/cloud_on.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+source $(dirname $0)/functions.sh
+
+cloud_init()
+{
+    # only ON is supported atm
+    CONFIG_DIR=${SECANT_CONFIG_DIR:-/etc/secant}
+    source ${CONFIG_DIR}/cloud.conf
+
+    export ONE_HOST ONE_XMLRPC
+}
+
+cloud_shutdown_vm()
+{
+    logging $TEMPLATE_IDENTIFIER "Shutting down VM $VM_ID." "DEBUG"
+    onevm shutdown --hard $VM_ID
+    if [ $? -ne 0 ]; then
+        logging $TEMPLATE_IDENTIFIER "Failed to shutdown VM $VM_ID." "ERROR"
+    fi
+}
+
+cloud_get_vm_ids()
+{
+    echo $(onevm list | awk '{ print $1 }' | sed '1d')
+    if [ $? -ne 0 ]; then
+        logging "Failed to get list of vm ids." "ERROR"
+    fi
+}
+
+cloud_get_template_ids()
+{
+    echo $(onevm template | awk '{ print $1 }' | sed '1d')
+    if [ $? -ne 0 ]; then
+        logging "Failed to get list of template ids." "ERROR"
+    fi    
+}
+
+cloud_vm_query()
+{
+    echo $(onevm show $VM_ID -x | xmlstarlet sel -t -v $query -n)
+    if [ $? -ne 0 ]; then
+        logging "Failed to read VM details with id $VM_ID." "ERROR"
+    fi
+}
+
+cloud_template_query()
+{
+    echo $(onevm show $TEMPLATE_ID -x | xmlstarlet sel -t -v $query -n)
+    if [ $? -ne 0 ]; then
+        logging "Failed to read VM details with id $TEMPLATE_ID." "ERROR"
+    fi
+}
+
+cloud_instantiate()
+{
+    echo $(onetemplate instantiate $TEMPLATE_ID $CTX_ADD_USER)
```

Comment:
TEMPLATE_ID needs to be a parameter, while $CTX_ADD_USER should be specified in the configuration